clients/client/src/main/java/org/projectnessie/client/api/PagingBuilder.java

```diff
@@ -15,16 +15,49 @@
  */
 package org.projectnessie.client.api;
 
-import org.projectnessie.model.EntriesResponse;
+import java.util.OptionalInt;
+import java.util.stream.Stream;
+import org.projectnessie.error.NessieNotFoundException;
 
-public interface PagingBuilder<R extends PagingBuilder<R>> {
+public interface PagingBuilder<R extends PagingBuilder<R, RESP, ENTRY>, RESP, ENTRY> {
 
-  /** Recommended: specify the maximum number of records to return. */
+  /**
+   * When using the manual paging, it is recommended to set this parameter, this parameter is
+   * configured when necessary via {@link #stream(OptionalInt)}.
+   *
+   * <p>Only for manual paging via {@link #get()} - do <em>not</em> call when using any of the
+   * {@link #stream()} functions.
```

Comment:
suggestion: `Sets the maximum number of records to be returned in a single response object from the get() method. When using stream() methods this parameter must not be set`.

Plus we'd error out if `stream()` is called _when_ this parameter is set. WDYT?

Reply:
At that point, it's an unfinished builder. But I've applied the javadoc suggestion.

Reply:
If mean in the `stream()` implementation we could check and throw an exception if there's a value for `maxRecords` in the builder. That would ensure the parameter is not misused with respect to `stream()`... WDYT?

Reply:
Alternatively, we could treat `maxRecords()` as a `.limit()` on the `stream()` too. I think it would still match current API semantics. The caller would just lose the paging token when using `stream()` as opposed to `get()`, but that should be self-evident, I guess.

Reply:
Oh my - good point. If a user fully consumes one stream, and then another one, it's fine. Concurrently... not so fine...